Week 4/Contest (Apr 3)/Hot Hike/elijah.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+partial class Program
+{
+  static void Main()
+  {
+    int n = int.Parse(Console.ReadLine());
+    var forecast = Console.ReadLine().Split(' ').Select(s => int.Parse(s)).ToArray();
+
+    int smallestTemp = -1;
+    int smallestIdx = -1;
+    
+    for (int i = 0; i < n - 2; i++)
+    {
+      int max = (int)Math.Max(forecast[i], forecast[i + 2]);
+
+      if (max < smallestTemp || smallestIdx == -1)
```

Comment:
This line lets you get away with setting the -1 value for smallest temp, but you can actually remove the smallestIndex check with a slight change here:
`int smallestTemp = -21`
```suggestion
      if (max < smallestTemp)
```
This way you always start with a default 'lowest' temperature, which gets overwritten after the start.